packages/material-ui/src/Tabs/Tabs.js

```diff
@@ -2,96 +2,214 @@ import * as React from 'react';
-    display: 'flex',
+const scrollbarSizeClassName = 'PrivateScrollbarSize';
+
+const useUtilityClasses = (styleProps) => {
+  const {
+    vertical,
+    fixed,
+    hideScrollbar,
+    scrollableX,
+    scrollableY,
+    flexContainerVertical,
+    centered,
+    scrollButtonsHideMobile,
+    classes,
+  } = styleProps;
+
+  const slots = {
+    root: ['root', vertical && 'vertical'],
+    scroller: [
+      'scroller',
+      fixed && 'fixed',
+      hideScrollbar && 'hideScrollbar',
+      scrollableX && 'scrollableX',
+      scrollableY && 'scrollableY',
+    ],
+    flexContainer: [
+      'flexContainer',
+      flexContainerVertical && 'flexContainerVertical',
+      centered && 'centered',
+    ],
+    indicator: ['indicator'],
+    scrollButtons: ['scrollButtons', scrollButtonsHideMobile && 'scrollButtonsHideMobile'],
+  };
+
+  return composeClasses(slots, getTabsUtilityClass, classes);
+};
+
+const TabsRoot = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'Root',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+
+      return deepmerge(
+        {
+          ...(styleProps.vertical && styles.vertical),
+          ...styles.scrollButtons,
+          ...(styleProps.scrollButtonsHideMobile && styles.scrollButtonsHideMobile),
+        },
+        styles.root || {},
+      );
+    },
   },
-  /* Styles applied to the flex container element if `orientation="vertical"`. */
-  flexContainerVertical: {
+)(({ styleProps, theme }) => ({
+  overflow: 'hidden',
+  minHeight: 48,
+  WebkitOverflowScrolling: 'touch', // Add iOS momentum scrolling.
+  display: 'flex',
+  ...(styleProps.vertical && {
     flexDirection: 'column',
-  },
-  /* Styles applied to the flex container element if `centered={true}` & `!variant="scrollable"`. */
-  centered: {
-    justifyContent: 'center',
-  },
-  /* Styles applied to the tablist element. */
-  scroller: {
-    position: 'relative',
-    display: 'inline-block',
-    flex: '1 1 auto',
-    whiteSpace: 'nowrap',
-  },
-  /* Styles applied to the tablist element if `!variant="scrollable"`. */
-  fixed: {
-    overflowX: 'hidden',
-    width: '100%',
-  },
-  /* Styles applied to the tablist element if `variant="scrollable"` and `orientation="horizontal"`. */
-  scrollableX: {
+  }),
+  ...(styleProps.scrollButtonsHideMobile && {
+    [`.${tabsClasses.scrollButtons}`]: {
+      [theme.breakpoints.down('sm')]: {
+        display: 'none',
+      },
+    },
+  }),
+  [`& .${scrollbarSizeClassName}`]: {
     overflowX: 'auto',
     overflowY: 'hidden',
+    // Hide dimensionless scrollbar on MacOS
+    scrollbarWidth: 'none', // Firefox
+    '&::-webkit-scrollbar': {
+      display: 'none', // Safari + Chrome
+    },
   },
-  /* Styles applied to the tablist element if `variant="scrollable"` and `orientation="vertical"`. */
-  scrollableY: {
-    overflowY: 'auto',
-    overflowX: 'hidden',
+}));
+
+const Scroller = experimentalStyled(
```

Comment:
AFAIK, the convention is:

```suggestion
const TabsScroller = experimentalStyled(
```

x the other